allennlp_models/pretrained.py

```diff
@@ -1,194 +1,46 @@
-from typing import Optional
+from typing import Dict
 
-from allennlp.models import load_archive
-from allennlp.predictors import Predictor, TextClassifierPredictor
+from allennlp.predictors import Predictor
 
-from allennlp_models.coref import CorefPredictor
-from allennlp_models.mc.predictors import TransformerMCPredictor
-from allennlp_models.pair_classification import DecomposableAttentionPredictor
-from allennlp_models.rc import ReadingComprehensionPredictor
-from allennlp_models.structured_prediction import (
-    SemanticRoleLabelerPredictor,
-    OpenIePredictor,
-    ConstituencyParserPredictor,
-    BiaffineDependencyParserPredictor,
-)
-from allennlp_models.tagging.predictors import SentenceTaggerPredictor
+from allennlp_models.classification import models as classification_models  # noqa: F401
+from allennlp_models.coref import models as coref_models  # noqa: F401
+from allennlp_models.generation import models as generation_models  # noqa: F401
+from allennlp_models.lm import models as lm_models  # noqa: F401
+from allennlp_models.mc import models as mc_models  # noqa: F401
+from allennlp_models.pair_classification import models as pc_models  # noqa: F401
+from allennlp_models.rc import models as rc_models  # noqa: F401
+from allennlp_models.structured_prediction import models as sp_models  # noqa: F401
+from allennlp_models.tagging import models as tagging_models  # noqa: F401
 
+from allennlp_models.common.model_card import get_pretrained_models
 
-# flake8: noqa: E501
+# TODO: rename all archive files to follow standard naming.
+# Standard: task-model.yyyy-mm-dd.tar.gz OR model-task.yyyy-mm-dd.tar.gz
 
+PRETRAINED_MODELS = get_pretrained_models()
```

Comment:
This is problematic. `model_card.py` might pick up more model cards after `pretrained.py` has been imported, but we've made a copy here, so we won't see those later updates anymore.

Why do we have to have this `dict` here anyways? Can you just read the needed information out of `model_card.py` directly in `get_predictor()`?

Reply:
I created the dict envisioning that we could pass it to the demo directly.

Reply:
Couldn't the demo just call `get_pretained_models()`? Or just access `model_card.PRETRAINED_MODELS`?

Reply:
Yes, it can. I wrote it this way so that pretrained.py remains the source of truth for everything outside, which is how it was originally meant to be. I put the global `PRETRAINED_MODELS` in `ModelCard` rather than pretrained.py only to simplify adding new pretrained models in the `ModelCard`'s init function itself. But we can technically do away with pretrained.py completely.